pkg/server/server.go

```diff
@@ -387,7 +387,8 @@ func (s *Server) Run(ctx context.Context) error {
 	// TODO: split apart everything after this line, into their own commands, optional launched in this process
 
 	controllerConfig := rest.CopyConfig(s.IdentityConfig)
-
+	// Setting the default restConfig timeout to 30 seconds to avoid any client is waiting infinitely
+	controllerConfig.Timeout = time.Second * 30
```

Comment:
Wondering, we should probably set that much earlier. IdentityConfig comes the loopback config if I am not mistaken. So maybe already the loopback config needs this timeout?